apollo-router-core/src/query_planner/router_bridge_query_planner.rs

```diff
@@ -38,12 +38,16 @@ impl QueryPlanner for RouterBridgeQueryPlanner {
 
         let planner_result = tokio::task::spawn_blocking(|| {
             plan::plan::<PlannerResult>(context, options.into())
-                .map_err(|e| QueryPlannerError::PlanningErrors(Arc::new(e)))
+                .map_err(QueryPlannerError::RouterBridgeError)
         })
-        .await??;
+        .await???;
 
         match planner_result {
-            PlannerResult::QueryPlan { node } => Ok(Arc::new(QueryPlan { root: node })),
+            PlannerResult::QueryPlan { node: Some(node) } => Ok(Arc::new(QueryPlan { root: node })),
+            PlannerResult::QueryPlan { node: None } => {
+                tracing::debug!("Empty query plan");
+                Err(QueryPlannerError::EmptyPlan)
```

Comment:
:grey_question: I think this is a good case for using `failfast_debug` instead?
```suggestion
                failfast_debug!("Empty query plan");
                Err(QueryPlannerError::EmptyPlan)
```

Reply:
https://github.com/apollographql/router/blob/13045c514ec9954f2c3713999c31b4b75fa805e9/apollo-router-core/src/lib.rs#L3-L12

Reply:
oh indeed!